app/assets/javascripts/discourse/app/components/directory-table.js

```diff
@@ -2,16 +2,88 @@ import Component from "@ember/component";
 import { action } from "@ember/object";
 
 export default Component.extend({
-  classNames: ["directory-table-container"],
+  lastScrollPosition: 0,
+  ticking: false,
+  topHorizontalScrollBar: null,
+  tableContainer: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+    this.set(
+      "tableContainer",
+      this.element.querySelector(".directory-table-container")
+    );
+    if (
+      this.tableContainer.getBoundingClientRect().bottom < window.innerHeight
+    ) {
+      // Bottom of the table is visible. Return and don't show top scrollbar
+      return;
+    }
+
+    this.set(
+      "topHorizontalScrollBar",
+      this.element.querySelector(".directory-table-top-scroll")
+    );
+    const fakeContent = this.topHorizontalScrollBar.querySelector(
+      ".directory-table-top-scroll-fake-content"
+    );
+    const table = this.tableContainer.querySelector(".directory-table");
+    fakeContent.style.width = `${table.offsetWidth}px`;
```

Comment:
What happens if the browser resizes? Does this need to be re-executed?

Reply:
People shouldn't do that

Reply:
Okay fine I'll take that into account

Reply:
Okay @eviltrout I refactored it a lot to capture all the elements in the `didInsertElement` then all the resizing is handled on resize window event